README
Name

    CatalystX::Usul - A base class for Catalyst MVC components

Version

    This document describes CatalystX::Usul version v0.17.$Rev: 1 $

Synopsis

       package MyApp::Controller;
    
       use Moose;
    
       BEGIN { extends qw(CatalystX::Usul::Controller) }
    
    
       package MyApp::Model;
    
       use Moose;
    
       extends qw(CatalystX::Usul::Model);

Description

    These modules provide a set of extension base classes for a Catalyst
    web application. Features include:

    Targeted at intranet applications

      The identity model supports multiple backend authentication stores
      including the underlying operating system accounts

    Thin controllers

      Most controllers make a single call to the model and so comprise of
      only a few lines of code. The interface model stashes data used by
      the view to render the page

    No further view programing required

      A single template toolkit instance is used to render all pages as
      either HTML or XHTML. The template forms one component of the "skin",
      the other components are: a Javascript file containing the use cases
      for the Javascript libraries, a primary CSS file with support for
      alternative CSS files, and a set of image files

      Designers can create new skins with different layout, presentation
      and behaviour for the whole application. They can do this for the
      example application, Munchies, whilst the programmers write the
      "real" application in parallel with the designers work

    Flexible development methodology

      These base classes are used by an example application, Munchies,
      which can be deployed to staging and production servers at the
      beginning of the project. Setting up the example application allows
      issues regarding the software technology to be resolved whilst the
      "real" application is being written. The example application can be
      deleted leaving these base classes for the "real" application to use

Subroutines/Methods

    This package is a placeholder for POD and contains no methods or
    functions

Configuration and Environment

    None

Diagnostics

    None

Dependencies

    Class::Usul

    To make the Captchas work GD::SecurityImage needs to be installed which
    has a documented dependency on libgd which should be installed first

Incompatibilities

    There are no known incompatibilities in this module

Bugs and Limitations

    There are no known bugs in this module. Please report problems to the
    address below. Patches are welcome

Acknowledgements

    Larry Wall - For the Perl programming language

Author

    Peter Flanigan, <pjfl@cpan.org>

License and Copyright

    Copyright (c) 2014 Peter Flanigan. All rights reserved

    This program is free software; you can redistribute it and/or modify it
    under the same terms as Perl itself. See perlartistic

    This program is distributed in the hope that it will be useful, but
    WITHOUT WARRANTY; without even the implied warranty of MERCHANTABILITY
    or FITNESS FOR A PARTICULAR PURPOSE



Changes
Revision history for CatalystX::Usul

0.17.1    2014-01-15 17:31:35
          - C:U:File::symlink class method replaced with a function
          - C:U:File::io class method replaced with a function
          - Apply trait for get_package_meta in ::CLI

0.16.1    2014-01-07 08:42:28
          - Moved ::Schema and ::TraitFor::ConnectInfo to Class::Usul

0.15.1    2014-01-06 12:49:48
          - Preparing to move ::Schema to Class::Usul
          - Updated toolchain. Drop prereqs below min perl ver

0.14.1    2013-11-23 14:25:54
          - Dropped dependency on MooX::Options
          - Bumped Class::Usul dependency to v0.33
          - Removed prefix and salt attr from ::Usul

0.13.1    2013-10-03 16:03:45
          - Updated git hooks
          - Bumped Class::Usul dependency

0.12.1    2013-09-29 13:12:34
          - Put strict back for Kwalitee
          - Added documentation strings to Schema
          - Added api_version to stash
          - Schema ensures schema classes loaded

0.11.1    2013-09-03 13:53:49
          - Bumped version dependencies
          - Using C::U::Exception not Unexpected
          - C::U::L10N::language renamed to locale

0.9.0     2013-08-19 20:31:01
          - Bumped version to reset patch number
          - Switched to Git from SVN

0.8.1320  2013-07-31
          - Raised minimum Perl version to v5.10.1
          - Updated toolchain
          - Internalized Moose exporter and constraints
          - Changed logname to loginid in ::Admin
          - Tightened ::TraitFor::ConnectInfo
          - Split out decrypt_from_cfg to Class::Usul
          - Updated inc::Bob to 1.14
          - Updated Moose dependency
          - CX::U::Schema::edit_credentials has bootstrap to create new ones
          - Fixed TraitFor::ConnectInfo visiting attributes in config object
          - Fixed bug in controller __want_app_closed function
          - Added redirect after execute option to action class
          - Updated inc::Bob to 1.6
          - Bumped version from 0.7 to 0.8
          - Added bsonid64 testing code to CLI
          - Uses PPI::HTML for Perl source syntax highlighting
          - C:U:Moose now imports CX:U:Constraints and MX:AS
          - Updated inc::Bob to 1.5
          - Added King's marin smoker to CPANTesting exceptions. See
            9800a69a-bc95-11e1-b285-d2fa227e6607. 1 fail and 167 passes
          - CX::U::Admin::_deploy_plack_server also copys env file to
            /etc/default
          - Fixed undefind value warning in Functions::app_prefix
          - Refactored to use Class::Usul::Moose;

0.7.1184  2012-04-21
          - Turned CPAN testing on. Sort::Naturally is installing again

0.7.1181  2012-04-17
          - F::DC::Constraints now using MX::Types. Updated CX::U:L10N
          - Eliminated exception_class method from CX::U::Base
          - CX::U::File now sets F::DC exception class via class method
          - Updated inc::Bob to 1.3
          - Removed IPC::SysV from Build.PL as it's unused
          - Changed domain name of cookies in CX::U::Controller since
            Firefox adds a leading dot
          - Moved assert_directory from CX::U::Programs to CX::U::Functions
          - Reduced POSIX imports in CX::U::IPC to WIFEXITED and WNOHANG
          - Swapped HTTP::DetectUserAgent for Parse::HTTP::UserAgent

0.6.1168  2012-04-04
          - Turning CPAN testing off because
            08289c92-7e3f-11e1-9d6f-f6dbfa7543f5. Thats 24hrs after peregrins
            Crixa upload and smokers are still failing because cpanmetadb has
            Sort::Naturally as not found

0.6.1168  2012-04-04
          - Fixed undefined warnings in Controller, View::HTML
            and CX::U::P::C::ModelHelper
          - Added a test of the controller begin method to 20controller.t
          - Refactored controller begin per request config loading

0.5.1164  2012-04-03
          - Added XML::NamespaceSupport as a dependency as Config::Any
            does not list it


Build.PL
# @(#)$Ident: Build.PL 2014-01-06 12:16 pjf ;

use strict;
use warnings;

use Module::Build 0.4004;

sub whimper { print {*STDOUT} $_[ 0 ]."\n"; exit 0 }

my $perl_ver = 5.010001;
my $name     = 'CatalystX-Usul'; $name =~ s{ - }{::}gmx;
my $prereqs  = {
  'build_requires' => {
    'Module::Build' => '0.4004',
    'Test::Requires' => '0.06',
    'version' => '0.88'
  },
  'configure_requires' => {
    'Module::Build' => '0.4004',
    'version' => '0.88'
  },
  'recommends' => {
    'App::cpanminus' => '1.6005',
    'Crypt::PassGen' => '0.05',
    'GD::SecurityImage' => '1.70',
    'IPC::Run' => '0.84',
    'Pod::ProjectDocs' => '0.36',
    'Syntax::Highlight::Perl' => '1.00'
  },
  'requires' => {
    'Catalyst' => '5.9002',
    'Catalyst::Action::RenderView' => '0.16',
    'Catalyst::Model::DBIC::Schema' => '0.48',
    'Catalyst::Plugin::Authentication' => '0.10016',
    'Catalyst::Plugin::ConfigComponents' => 'v0.8.0',
    'Catalyst::Plugin::ConfigLoader' => '0.19',
    'Catalyst::Plugin::InflateMore' => 'v0.7.0',
    'Catalyst::Plugin::Session' => '0.31',
    'Catalyst::Plugin::Session::State::Cookie' => '0.17',
    'Catalyst::Plugin::Session::Store::FastMmap' => '0.14',
    'Catalyst::Plugin::Static::Simple' => '0.30',
    'Catalyst::Plugin::StatusMessage' => '1.002000',
    'Catalyst::View::TT' => '0.36',
    'CatalystX::Component::Traits' => '0.16',
    'Class::Load' => '0.18',
    'Class::Null' => '2.101420',
    'Class::Usul' => 'v0.36.0',
    'Crypt::Eksblowfish' => '0.009',
    'DBIx::Class' => '0.08009',
    'Data::Pageset' => '1.06',
    'Data::Serializer' => '0.53',
    'Data::Validation' => 'v0.14.0',
    'Email::MIME' => '1.906',
    'Email::Sender' => '0.093380',
    'Encode' => '2.12',
    'Encode::Guess' => '2.00',
    'File::DataClass' => 'v0.32.0',
    'File::Gettext' => 'v0.22.0',
    'File::MailAlias' => 'v0.19.0',
    'File::UnixAuth' => 'v0.19.0',
    'HTML::FillInForm' => '2.00',
    'HTML::FormWidgets' => 'v0.21.0',
    'HTML::Parser' => '3.69',
    'HTTP::Date' => '6.02',
    'HTTP::Message' => '6.06',
    'IO::Interactive' => 'v0.0.6',
    'Imager' => '0.90',
    'Import::Into' => '1.001000',
    'JSON' => '2.50',
    'MIME::Types' => '1.31',
    'Moo' => '1.003000',
    'Moose' => '2.0802',
    'MooseX::AttributeShortcuts' => '0.014',
    'MooseX::Types' => '0.35',
    'MooseX::Types::Common' => '0.001008',
    'MooseX::Types::LoadableClass' => '0.008',
    'PPI' => '1.215',
    'PPI::HTML' => '1.08',
    'Parse::HTTP::UserAgent' => '0.33',
    'SQL::Translator' => '0.11016',
    'Sub::Exporter' => '0.982',
    'Template' => '2.22',
    'Time::Elapsed' => '0.29',
    'TryCatch' => '1.003000',
    'XML::Simple' => '2.18',
    'namespace::autoclean' => '0.13',
    'namespace::sweep' => '0.006',
    'perl' => '5.010001'
  },
};

if ($] < $perl_ver) {
   $ENV{AUTOMATED_TESTING}
      or $ENV{PERL_CPAN_REPORTER_CONFIG}
      or $ENV{PERL5_CPANPLUS_IS_VERSION}
      or whimper "Minimum required Perl version is ${perl_ver}";
   $prereqs->{build_requires} = { %{ $prereqs->{configure_requires} } };
   $prereqs->{build_requires}->{ 'Test::Requires' } = 0.06;
   delete $prereqs->{requires};
}

Module::Build->new(
  'module_name' => $name,
  'notes'       => { min_perl_version => $perl_ver },
  %{ $prereqs },

)->create_build_script;

exit 0;
